offset.go

```diff
@@ -0,0 +1,414 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/user"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+)
+
+type offsetConfig struct {
+	flags     *flag.FlagSet
+	brokers   []string
+	group     string
+	topic     *regexp.Regexp
+	partition int32
+	setOldest bool
+	setNewest bool
+	set       int64
+	verbose   bool
+	version   sarama.KafkaVersion
+	args      struct {
+		brokers   string
+		group     string
+		topic     string
+		partition int
+		set       string
+		verbose   bool
+		version   string
+	}
+}
+
+type offsets struct {
+	ConsumerGroup   string `json:"consumer-group,omitempty"`
+	Topic           string `json:"topic"`
+	Partition       int32  `json:"partition"`
+	PartitionOffset int64  `json:"partition-offset"`
+	ConsumerOffset  *int64 `json:"consumer-offset,omitempty"`
```

Comment:
Because it prints the offsets and I wanted it to be printed when the int was 0 but omitted when it was actually empty (no consumer group offsets, if we are displaying only offsets for partitions).

The pointer allowed me to omit the empty ConsumerOffset, while still allowing me to print a zero value int